### PR TITLE
Add workshop box for teachers with RPs that don't offer CSA

### PR DESCRIPTION
**Pull request by megcrenshaw**
Currently, if CSA workshops are available in their region, teachers will see this
image

If CSA workshops are *not* available in their region, teachers see this
image

With the new functionality, teachers see this if there are no CSA workshops (screenshot of local environment––I didn't set RP information for `appsPriorityDeadlineDate` so no call to action):
Screen Shot 2022-03-02 at 4 10 53 PM

And nothing changes if there are CSA workshops (screenshot of local environment––I didn't set RP information for `appsPriorityDeadlineDate` so no call to action):
Screen Shot 2022-03-02 at 4 14 37 PM

## Links

- spec: [Alerting Users of Code.org CSA Workshop Solution](https://docs.google.com/document/d/1dI8DZJ0SH01KCyreEmMLbO3Q7OmpvmIBEUfamXPZ5RQ/edit)
- jira ticket: [PLAT-1574](https://codedotorg.atlassian.net/browse/PLAT-1574)

## Follow-up work

I created a follow-up ticket for test coverage: https://codedotorg.atlassian.net/browse/PLAT-1650.

Description:
The RegionalPartnerSearch has 0 test coverage. Right now, trying to add test coverage results in errors about how we are setting state in the component:

`Can't call %s on a component that is not yet mounted. This is a no-op, but it might indicate a bug in your application. Instead, assign to `this.state` directly or define a `state = {};` class property with the desired state in the %s component.
`

We should probably refactor this component to use hooks, which would allow us to avoid mounting the component to test it.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/RegionalPartnerSearch.jsx [hunk 1]
```diff
@@ -136,27 +137,27 @@ class RegionalPartnerSearch extends Component {
 
     let workshopCollections = [
       {
-        heading: 'CS Discoveries Workshops',
+        heading: `${ActiveCourseWorkshops.CSD} Workshops`,
         workshops:
           partnerInfo &&
           partnerInfo.summer_workshops.filter(
-            workshop => workshop.course === 'CS Discoveries'
+            workshop => workshop.course === `${ActiveCourseWorkshops.CSD}`
```
Comment (tim-dot-org) [inline apps/src/templates/RegionalPartnerSearch.jsx]: tiny nit: these are already strings, so they don't need to be wrapped

**Review by tim-dot-org — APPROVED** [review]

File: apps/src/templates/RegionalPartnerSearch.jsx (hunk 1 above)
Comment (tim-dot-org) [reply apps/src/templates/RegionalPartnerSearch.jsx]: otherwise, looks great!

File: apps/src/templates/RegionalPartnerSearch.jsx [hunk 2]
```diff
@@ -303,9 +304,32 @@ class RegionalPartnerSearch extends Component {
                   collection => collection.workshops.length === 0
                 ) && <div>Workshop details coming soon!</div>}
```
Comment (tim-dot-org) [inline apps/src/templates/RegionalPartnerSearch.jsx:305]: maybe a weird quirk, if there are no workshops at all, it'll show this text and also the CSA text

Comment (megcrenshaw) [reply apps/src/templates/RegionalPartnerSearch.jsx:305]: Strong point––I'll check with Tess what we want to do in that situation

Comment (megcrenshaw) [reply apps/src/templates/RegionalPartnerSearch.jsx:305]: She says it's not a huge deal since this won't happen next year, but I'll add something for it

Comment (megcrenshaw) [reply apps/src/templates/RegionalPartnerSearch.jsx:305]: Just tweaked it––new screenshots for workshops, and then if there are no workshops
Screen Shot 2022-03-04 at 2 21 53 PM
Screen Shot 2022-03-04 at 2 22 46 PM

**Review by tim-dot-org — APPROVED** [review]
perfect!

**Pull request merged**